doc/user/load_save.md

```diff
@@ -1,61 +1,573 @@
+   * Load `X` from the given file `filename` with default options:
+     - the format of the file is [auto-detected](#formats) based on the
+       extension of the file, and
+     - an exception is *not* thrown on an error.
+   * Returns a `bool` indicating whether the load was a success.
+   * `X` can be [any supported load type](#types).
+
+ - `data::Load(filenames, X)`
+   * Load `X` from a set of  file `filenames` with default options:
+     - the format of the all files is [auto-detected](#formats) based on the
+       extension of the file, and
+     - an exception is *not* thrown on an error.
+   * Returns a `bool` indicating whether the load was a success.
+   * `X` can be [any supported load type](#types).
+
+ - `data::Load(filename, X, Option1 + Option2 + ...)`
+   * Load `X` from the given file `filename` with the given options.
+   * Returns a `bool` indicating whether the load was a success.
+   * `X` can be [any supported load type](#types).
+   * The given options must be from the
+     [list of standalone operators](#dataoptions) and be appropriate for the type
+     of `X`.
+
+ - `data::Load(filename, X, opts)`
+   * Load `X` from the given file `filename` with the given options specified in `opts`.
+   * Returns a `bool` indicating whether the load was a success.
+   * `X` can be [any supported load type](#types).
+   * `opts` is a [`DataOptions` object](#dataoptions) whose subtype matches the
+     type of `X`.
+
+***Note:*** when loading a set of files into one matrix `X`, all files should have
+the same format and number of data points internally.  See [image data](#image-data) for more
+details.
+
+***Note:*** when loading images, it is possible to load
+multiple images into one matrix `X`.  See [image data](#image-data) for more
+details.
 
-## Numeric data
+---
 
-Numeric data or general numeric matrices can be loaded or saved with the
-following functions.
+*Simple example:*
 
- - `data::Load(filename, matrix, fatal=false, transpose=true, format=FileType::AutoDetect)`
- - `data::Save(filename, matrix, fatal=false, transpose=true, format=FileType::AutoDetect)`
-   * `filename` is a `std::string` with a path to the file to be loaded.
+```c++
+// See https://datasets.mlpack.org/iris.csv.
+arma::mat x;
+mlpack::data::Load("iris.csv", x);
 
-   * By default the format is auto-detected based on the file extension, but can
-     be explicitly specified with `format`; see [Formats](#formats).
+std::cout << "Loaded iris.csv; size " << x.n_rows << " x " << x.n_cols << "."
+    << std::endl;
+```
 
-   * `matrix` is an `arma::mat&`, `arma::Mat<size_t>&`, `arma::sp_mat&`, or
-     similar (e.g., a reference to an Armadillo object that data will be loaded
-     into or saved from).
+Among other things, the file format can be easily specified:
 
-   * If `fatal` is `true`, a `std::runtime_error` will be thrown on failure.
+```c++
+// See https://datasets.mlpack.org/iris.csv.
+arma::mat x;
+mlpack::data::Load("iris.csv", x, CSV);
```

Comment:
```suggestion
mlpack::data::Load("iris.csv", x, mlpack::data::CSV);
```
I suspect a bunch of the other examples fail for reasons like this.  As an aside: do you think all of the shorthand options should be in just `mlpack::`?  Or should we keep them in `mlpack::data::`?  I am ok with either, whatever you think is better.